### ISSUE TITLE
ServerDataHelper.searchrecords(OFieldsHelper fields, ODomain domain, int limit) is broken

### ISSUE THREAD

**Issue by PipoloyJo** [Odoo-mobile/framework#48]
Hey guys,

We've forked your project and worked on a new module for Delivery Orders (stock.picking.out),

Each DO is linked to multiple Stockmoves (stock.move) through a OneToMany relation with the field moves_lines in stock.picking.out. 

We are implementing a live request feature as a fallback if the DO has not been synced on device.
I stumbled upon a helper class, ServerDataHelper which has a search_records method.

It performs a search_read on the server. This is exactly what we want. But I found that when converting the JSON returned to a valid ODataRow some stuff was broken.

```
public List<ODataRow> searchRecords(OFieldsHelper fields, ODomain domain, int limit) {
    List<ODataRow> items = new ArrayList<ODataRow>();
    try {
         if (mApp.inNetwork()) {
             JSONObject result = mOdoo.search_read(mModel.getModelName(),
                    fields.get(), domain.get(), 0, limit, null, null);
            JSONArray records = result.getJSONArray("records");
            if (records.length() > 0) {
                for (int i = 0; i < records.length(); i++) {
                    items.add(JSONUtils.toDataRow(records.getJSONObject(i)));
                }
            }
        }
    } catch (Exception e) {
        e.printStackTrace();
    }
    return items;
}
```

The OneToMany relation is not accounted for, and when trying to access the stockmoves linked by the field move_lines It just crashes saying  it cannot convert a JSONArray to a List of ODataRow

This is the faulty method:

```
public static ODataRow toDataRow(JSONObject json) {
    ODataRow row = new ODataRow();
    try {
    @SuppressWarnings("unchecked")
        Iterator<String> keys = json.keys();
        while (keys.hasNext()) {
            String key = keys.next();
            row.put(key, json.get(key));
        }
    } catch (Exception e) {
        e.printStackTrace();
    }
    return row;
}
```

So basically we end up with a JSONArray in the move_lines field of our Delivery Order.
I'm gonna try to fix it, but this is kind of bad, since it can return none-usable object without warning.

If you guys have any solution in the meantime, I'll be happy to hear about it. I'm trying to bypass the syncWithServer stuff since it seems quite long and not fully optimized.

Thanks,
